packages/layout/src/tree-layout/drawing-manager/drawingManager.js

```diff
@@ -0,0 +1,127 @@
+import { HTMLRenderer } from '../renderers/html-renderer';
+import { Utils } from '../utils/utils';
+
+export class DrawingManager {
+    constructor (data, renderer, container) {
+        this.data = data.tree;
+        this.componentMap = data.componentMap;
+        this.layoutClassName = data.layoutClassName;
+        this.renderer = renderer;
+        global.__renderer = renderer; // TODO change global into diff place
+        this.mount = Utils.isDOMElement(container) ? container : Utils.getElement(container);
+    }
+
+    _drawLayout () {
+        switch (this.renderer) {
+        case 'html' :
+            this.renderHTML();
+            break;
+        default:
+            break;
+        }
+    }
+
+    _drawComponent (componentData) {
+        componentData.children.forEach((node) => {
+            if (node.model && node.model.host) {
```

Comment:
node.model can be stored in a variable instead of being referenced multiple times